File: app/src/main/java/com/example/jarsical/fragments/library/LibraryAdapter.java
package com.example.jarsical.fragments.library;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.jarsical.PlayPlaylistActivity;
import com.example.jarsical.PlaySongActivity;
import com.example.jarsical.R;
import com.example.jarsical.Song;
import com.squareup.picasso.Picasso;

import java.util.List;

public class LibraryAdapter extends RecyclerView.Adapter<LibraryAdapter.ViewHolder> {
    public LibraryAdapter(List<Song> playlist) {
        this.playlist = playlist;
    }
    List<Song> playlist;


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.custom_row_library,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        if(playlist.size()>=1){
            Song song = playlist.get(position);
            holder.songTitle.setText(song.getTitle());
            holder.songArtist.setText(song.getArtists());
            Picasso.get().load(song.getArtLink()).fit().placeholder(R.drawable.ic_baseline_error_24).into(holder.imageButton);//Uses picasso to download images into imageButton
            holder.imageButton.setContentDescription(song.getId());

            holder.imageButton.setOnClickListener(v -> {
                Intent intent = new Intent(v.getContext(), PlayPlaylistActivity.class);
                intent.putExtra("index",position);
                v.getContext().startActivity(intent);
            });
        }else{
        }
    }

    @Override
    public int getItemCount() {
        if(playlist == null){
            return 0;
        } else {
            return playlist.size();
        }

    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView songTitle, songArtist;
        ImageButton imageButton;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            songTitle = itemView.findViewById(R.id.txtSongName);
            songArtist = itemView.findViewById(R.id.txtSongArtist);
            imageButton = itemView.findViewById(R.id.imageButton);

        }
    }
}

File: app/src/main/java/com/example/jarsical/fragments/library/libraryAdapter.kt
package com.example.jarsical.fragments.library

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.jarsical.R
import com.example.jarsical.R.layout.custom_row2
import com.example.jarsical.Song
import com.example.jarsical.fragments.library.libraryAdapter.MyViewHolder
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.custom_row.view.*

class libraryAdapter(favList: ArrayList<Song>): Adapter<MyViewHolder>()  {

    var favList: ArrayList<Song> = ArrayList()

    class MyViewHolder(itemView: View): ViewHolder(itemView){
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        return MyViewHolder(LayoutInflater.from(parent.context).inflate(custom_row2, parent, false))

    }

    override fun getItemCount(): Int {
        return favList.size
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        var currentItem = favList[position]

        holder.itemView.txtSongName.text = currentItem.title
        holder.itemView.txtSongArtist.text = currentItem.artists
        holder.itemView.txtSongLength.text = currentItem.songLength.toString()

        //picasso allows us to use cover art without downloading
        Picasso.get().load(currentItem.artLink).placeholder(R.drawable.ic_baseline_error_24).into(holder.itemView.imageButton)// loads images into imageButton
        //sets content description
        holder.itemView.imageButton.contentDescription = currentItem.id
        val test = currentItem.id
        Log.d("libraryAdapter",test)

    }

    fun setData(song: ArrayList<Song>){
        this.favList = song
        notifyDataSetChanged()
        Log.d("setData",favList.size.toString())

    }
}